src/coreclr/jit/fgopt.cpp

```diff
@@ -2950,7 +2950,10 @@ bool Compiler::fgOptimizeEmptyBlock(BasicBlock* block)
                         else
                         {
                             Statement* nopStmt = fgNewStmtAtEnd(block, nop);
-                            fgSetStmtSeq(nopStmt);
+                            if (fgNodeThreading == NodeThreading::AllTrees)
+                            {
+                                fgSetStmtSeq(nopStmt);
+                            }
```

Comment:
This function can be invoked before we are keeping nodes threaded so in those cases there is no need to thread them.